src/controllers/controller.bar.js

```diff
@@ -86,6 +86,10 @@ defaults._set('horizontalBar', {
 	}
 });
 
+function getValue(scale, dataPoint) {
+	return typeof dataPoint === 'string' ? Number(dataPoint) : scale.getRightValue(dataPoint);
```

Comment:
I tried that and it seemed to break the tests. I also think it's rather silly that people would pass in a number as a string and would just as much consider deprecating support for that in 3.0 so wasn't sure I wanted to encourage to be done for even more scales